File: ios/Doglyad/Domain/Ultrasound/USExaminationData.swift
import DoglyadDatabase
import Foundation

struct USExaminationData: Codable {
    let usExaminationTypeId: String
    let photos: [USExaminationScanPhoto]
    let examinationNumber: String
    let patientName: String
    let patientGender: PatientGender
    let patientDateOfBirth: Date
    let patientHeight: Double?
    let patientWeight: Double?
    let patientComplaints: String?
    let examinationDescription: String
}

extension USExaminationData {
    static func fromDB(
        _ db: USExaminationDataDB
    ) -> USExaminationData {
        USExaminationData(
            usExaminationTypeId: db.usExaminationTypeId,
            photos: db.photos.map { USExaminationScanPhoto.fromDB($0) },
            examinationNumber: db.examinationNumber,
            patientName: db.patientName,
            patientGender: PatientGender(rawValue: db.patientGenderRawValue) ?? .male,
            patientDateOfBirth: db.patientDateOfBirth,
            patientHeight: db.patientHeight,
            patientWeight: db.patientWeight,
            patientComplaints: db.patientComplaints,
            examinationDescription: db.examinationDescription
        )
    }

    func toDB() -> USExaminationDataDB {
        USExaminationDataDB(
            usExaminationTypeId: usExaminationTypeId,
            photos: photos.map { $0.toDB() },
            examinationNumber: examinationNumber,
            patientName: patientName,
            patientGenderRawValue: patientGender.rawValue,
            patientDateOfBirth: patientDateOfBirth,
            patientHeight: patientHeight,
            patientWeight: patientWeight,
            patientComplaints: patientComplaints,
            examinationDescription: examinationDescription
        )
    }
}

File: backend/main/app/model/ultrasound/us_examination_data.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, field_validator

from app.model.ultrasound.us_examination_scan_photo import USExaminationScanPhoto


class USExaminationData(BaseModel):
    usExaminationTypeId: str
    photos: list[USExaminationScanPhoto]
    examinationNumber: str
    patientName: str
    patientGender: str
    patientDateOfBirth: datetime
    patientHeight: float
    patientWeight: float
    patientComplaint: str | None = None
    examinationDescription: str

    @field_validator("patientComplaint")
    @classmethod
    def normalize_optional_complaint(cls, value: str | None) -> str | None:
        if value is None:
            return None
        return value.strip() or None
